GB3 homework 5

emikhailow opened the pull request:
GB3 homework 5

emikhailow committed:
GB3 homework 5

emikhailow committed:
GB3 homework 5 (внес исправления)

emgusev commented on src/main/java/homework5/Car.java:
```diff
@@ -0,0 +1,49 @@
+            System.out.println(this.name + " готовится");
+            Thread.sleep(500 + (int)(Math.random() * 800));
+            System.out.println(this.name + " готов");
+            MainClass.startLatch.countDown();
```
Лучше через метод получать поля. MainClass.getStartLatch()

emgusev commented on src/main/java/homework5/Car.java:
```diff
@@ -0,0 +1,49 @@
+        }
+        for (int i = 0; i < race.getStages().size(); i++) {
+            race.getStages().get(i).go(this);
+        }
```
Формально тут с использованием лока можно еще победителя определить.

emikhailow committed:
GB3 homework 5
Учел замечания:
1) Лучше через метод получать поля. MainClass.getStartLatch()
2) Формально тут с использованием лока можно еще победителя определить
